X86 non deep sx power seq

vphirema opened the pull request.

vphirema committed:
Power: Add GPIO init code for Non-DSx power sequencing

Added basic code to initialize GPIOs of power sequence the Intel x86
Non deep sleep well platforms under subsystem folder.

Signed-off-by: Vijay Hiremath <vijay.p.hiremath@intel.com>

vphirema committed:
espi: Add eSPI configuration for Non-DSX platforms

Added code to enable eSPI communication to support the power
sequencing of Non Deep Sleep Well x86 platforms.

Signed-off-by: Vijay Hiremath <vijay.p.hiremath@intel.com>

jackrosenthal commented on subsys/power_seq/x86_non_dsx.h:
```diff
@@ -0,0 +1,62 @@
+#define POWER_EC_PCH_RSMRST_DELAY_MS	10
+#define POWER_EC_PCH_SYS_PWROK_DELAY_MS	50
+#define POWER_EC_VR_EN_VCCIN_DELAY_MS	5
+#define POWER_EC_PCH_PM_PWRBTN_DELAY_MS	200
```
should these be kconfig options?

vphirema replied:
thank you for the feedback Jack.
These are constants needed by PCH, they don't change.

sjg20 commented:
How do we call / configure this code? Is there an example?

jettr commented on subsys/power_seq/x86_non_dsx.c:
```diff
@@ -0,0 +1,141 @@
+static void enable_power_rails(bool enable)
+{
+	if (enable) {
+		gpio_set_level(AP_5P0V_A_RAIL, 1);
```
We should probably define these gpios in a power-sequence device in the device tree. This code would use that device information and expect certain gpio names to be populated in the device tree, like en-ap-5p0v-a. Then this cod would look that up and use the gpio_* Zephyr API (e.g `gpio_pin_get_raw` `gpio_port_set_bits_raw` and `gpio_port_clear_bits_raw`) to interact with the gpios.

vphirema replied:
> How do we call / configure this code? Is there an example?

Added sample code

vphirema replied:
> We should probably define these gpios in a power-sequence device in the device tree. This code would use that device information and expect certain gpio names to be populated in the device tree, like en-ap-5p0v-a. Then this cod would look that up and use the gpio_* Zephyr API (e.g `gpio_pin_get_raw` `gpio_port_set_bits_raw` and `gpio_port_clear_bits_raw`) to interact with the gpios.

ACK, added device tree now, I will work on adding follow-on CLs. Try to learn this new interface

jettr commented on subsys/espi/ndsx_espi.c:
```diff
@@ -0,0 +1,106 @@
+	}
+}
+
+static void espi_bus_handler(const struct device *dev,
```
The code in this function seems generic to eSPI. I would expect this to be in the espi driver itself

vphirema replied:
ACK, I will add in follow on CLs

jettr commented on subsys/espi/ndsx_espi.c:
```diff
@@ -0,0 +1,106 @@
+	}
+}
+
+static void espi_vw_handler(struct espi_event *event)
```
The code in this function sees like it it target to the power sequencing part. I think this code should go in the power sequence folder (in the same file as above espi power sequencing stuff)

vphirema replied:
ACK

jettr commented on subsys/espi/ndsx_espi.c:
```diff
@@ -0,0 +1,106 @@
+static struct espi_callback espi_chan_cb;
+static struct espi_callback espi_vw_cb;
+
+static void espi_bus_reset(void)
```
The code in this function sees like it it target to the power sequencing part. I think this code should go in the power sequence folder (maybe as a separate file that has to do with eSPI though)

vphirema replied:
ACK

jettr commented on subsys/espi/ndsx_espi.c:
```diff
@@ -0,0 +1,106 @@
+
+u8_t vw_get_level(enum espi_vwire_signal signal)
+{
+	u8_t level;
```
```suggestion
	uint8_t level;
```
FYI, Zephyr is trying to more to the C99 types, so unit8_t and the like

vphirema replied:
ACK

vphirema committed:
board: Add Volteer Google Chrome Reference Design board

Added basic Embeded Controller code for Volteer (Google Chrome
Reference Design board) with Nuvoton EC and Intel Tigerlake SOC.

Signed-off-by: Vijay Hiremath <vijay.p.hiremath@intel.com>

vphirema committed:
x86 Power: Add power up sequence for Non Deep Sx platforms

Added basic code to do power up sequence of Intel x86 Non deep
sleep well platforms.

Signed-off-by: Vijay Hiremath <vijay.p.hiremath@intel.com>

vphirema committed:
x86: Add passthrough handler for Non Deep Sx

Added code to handle passthrough signals of the PCH via EC for
non Deep Sleep well platforms.

vphirema committed:
GPIO: Move GPIO flags to a separate header

Moved GPIO flags to a separate header so that it can be called
by devicetree GPIO include file without build errors.

Signed-off-by: Vijay Hiremath <vijay.p.hiremath@intel.com>

jettr commented on include/power/ndsx_espi.h:
```diff
@@ -0,0 +1,22 @@
+/**
+ * @brief Return Virtual wire value
+
+ * @retval Acatual Virtual Wire value or 0 on failure
```
s/Acatual/Actual/

Should we return an int and use a negative return to indicate an error instead?

vphirema replied:
ACK, will change it

jettr commented on include/power/x86_non_dsx.h:
```diff
@@ -104,6 +104,9 @@ enum power_states_ndsx {
 #define POWER_EC_VR_EN_VCCIN_DELAY_MS	5
+#define POWER_EC_PCH_PM_PWRBTN_DELAY_MS	200
+
+void espi_bus_reset(void);
```
The prefix for this file `non_dsx.h` doesn't match the other in this CL : `ndsx`

vphirema replied:
ACK, will change it

jettr commented on subsys/power/x86_non_dsx.c:
```diff
@@ -244,11 +239,24 @@ static void power_pass_thru_handler(const char *in_signal,
+	if (gpio_get_level(GPIO_NET_NAME(PCH_EC_SLP_SUS_L))) {
+		LOG_INF("Toggle PM PWRBTN");
+
+		gpio_set_level(GPIO_NET_NAME(EC_PCH_PWR_BTN_ODL), 0);
```
This is where we should be using device tree to reference the gpios that the power sequence node listed

jettr commented on include/dt-bindings/gpio/gpio.h:
```diff
@@ -7,6 +7,8 @@
 #ifndef ZEPHYR_INCLUDE_DT_BINDINGS_GPIO_GPIO_H_
 #define ZEPHYR_INCLUDE_DT_BINDINGS_GPIO_GPIO_H_
 
+#include <drivers/gpio_flags.h>
```
So I have been wanting a similar change to Zephyr, but I think we need to get a more general concensus on making this change. The `gpio_dt_flags_t` is only 8-bits, so it does seem by design. We should open a discussion about this

pabigot replied:
This is indeed by design: Zephyr only allows devicetree use of the specifier flags that are supported in devicetree files by Linux.  I believe the intent is to retain devicetree source file compatibility independent of OS.  Best to open an RFC issue if you want that changed.

vphirema replied:
ACK, i will file a RFC

vphirema replied:
submitted an RFC, I don't have access to add people 
https://github.com/zephyrproject-rtos/zephyr/issues/29908

jettr commented on boards/arm/volteer/Kconfig.board:
```diff
@@ -0,0 +1,6 @@
+# Copyright (c) 2020 Intel Corporation
+# SPDX-License-Identifier: Apache-2.0
+
+config BOARD_VOLTEER
```
You are your thoughts about working on the downstream version of volteer for now until we get things working. Then we will definitely want to upstream this board.

vphirema replied:
ACK, I will delete this CL

andrewboie commented:
can you update the PR description and title with a concise summary of what this pull request is doing?

vphirema commented:
> can you update the PR description and title with a concise summary of what this pull request is doing?
I have added a feature Feature request 
https://github.com/zephyrproject-rtos/zephyr/issues/29909

andrewboie commented:
> can you update the PR description and title with a concise summary of what this pull request is doing?
> I have added a feature Feature request
> #29909

No, I'm serious. What you have right now still reads:

> No description provided.

When submitting PRs please provide a paragraph or two summarizing what this commit changes, why the changes were made, and links to any supporting GH issues.

The title of the PR should also be descriptive and not just be whatever the (often heavily abbreviated) branch name is.

You are able to edit both the description and title of PRs even after they are submitted. Please do this. Sending PRs with empty description is unacceptable and is burdensome to reviewers.

vphirema commented:
> > can you update the PR description and title with a concise summary of what this pull request is doing?
> > I have added a feature Feature request
> > #29909
> 
> No, I'm serious. What you have right now still reads:
> 
> > No description provided.
> 
> When submitting PRs please provide a paragraph or two summarizing what this commit changes, why the changes were made, and links to any supporting GH issues.
> 
> The title of the PR should also be descriptive and not just be whatever the (often heavily abbreviated) branch name is.
> 
> You are able to edit both the description and title of PRs even after they are submitted. Please do this. Sending PRs with empty description is unacceptable and is burdensome to reviewers.

ACK. I haven't modified the code/commit yet. Do I need to wait until the Feature request gets some attention or can submit the code before that?
